extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/exemplars/OpenTelemetryExemplarContextUnwrapper.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.micrometer.runtime.export.exemplars;
+
+import java.util.function.Function;
+
+import jakarta.enterprise.context.Dependent;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.quarkus.opentelemetry.runtime.QuarkusContextStorage;
+
+@Dependent
+public class OpenTelemetryExemplarContextUnwrapper implements OpenTelemetryContextUnwrapper {
+
+    @Override
+    public <P, R> R executeInContext(Function<P, R> methodReference, P parameter, io.vertx.core.Context requestContext) {
+        if (requestContext == null) {
+            return methodReference.apply(parameter);
+        }
+
+        Context newContext = QuarkusContextStorage.getContext(requestContext);
+
+        if (newContext == null) {
```

Comment:
IS it really possible? Or should it be considered as a failure (or at least logged)

Reply:
It will be checked in the `attach` method - but I'm wondering if we need to do it before.

Reply:
It is totally possible when the metrics SPI method is invoked directly from the EL thread (e.g. when reporting bytes sent metrics)